bin/csv_from_oai.rb

```diff
@@ -0,0 +1,80 @@
+require 'csv'
+require 'oai'
+require 'byebug'
+
+# Input for dynamic email
+puts "Enter your email address:"
+email = gets.chomp
+
+# OAI client setup
+client = OAI::Client.new(
+  "http://oai.adventistdigitallibrary.org/OAI-script",
+  headers: { from: email },
+  parser: 'libxml'
+)
+
+opts = {
+  metadata_prefix: "oai_adl"
+}
+
+CSV.open('csv_from_oai.csv', 'wb') do |csv|
+  # Write the headers to the CSV file
+  csv << ['set', 'identifier', 'derivative_type', 'URL']
+
+  sets = ["adl:thesis",
+    "adl:periodical",
+    "adl:issue",
+    "adl:article",
+    "adl:image",
+    "adl:book",
+    "adl:other"
+  ]
+
+  sets.each do |set|
+    records = client.list_records(opts.merge(set: set))
+    records.each do |r|
+      identifier = r.header.identifier
+      thumbnail_urls = r.metadata.first.find('thumbnail_url').map(&:content)
+      related_urls = r.metadata.first.find('related_url').map(&:content)
+
+      thumbnail_urls.each do |thumbnail_url|
+        # Split thumbnail_url and extract file type
+        if thumbnail_url.include?(';')
+          thumbnail_url, query_string = thumbnail_url.split(';')
```

Comment:
Talk to me about this line.  Why the variable name `query_string`?  The query string is what follows a `?` in a URL.  So I'm confused why you're splitting on a `;` and calling one of the variables a query string.

Reply:
I did this as a means of splitting the files from the URLs based on the fact two remote files were writing to the same line.

Reply:
What were you seeing in the thumbnail_url element that had the `;`?  My suspicion is that you didn't have a query string to the right of the first `;` but instead that you had multiple URLs.  As in multiple thumbnails.

Note this also applies to the other section that has `query_string`

Reply:
![image](https://user-images.githubusercontent.com/95306716/233146842-05d9fc7c-6148-4a02-a423-3b22bc847eb2.png)
I have been trying to figure out a solution for splitting " pdf ; txt "